designs/sam_build_cmd.rst

```diff
@@ -0,0 +1,442 @@
+| Action      | Command                                           |
++=============+===================================================+
+| Resolve     | ``dep ensure -v``                                 |
++-------------+---------------------------------------------------+
+| Compile     | ``GOOS=linux go build  -ldflags="-s -w" main.go`` |
++-------------+---------------------------------------------------+
+| Copy Source | No Op                                             |
++-------------+---------------------------------------------------+
+
+
+Dotnet using Dotnet CLI
+^^^^^^^^^^^^^^^^^^^^^^^
+
+**Manifest Name**: ``*.csproj``
+
++-------------+----------------------------------------------------------------------------------------+
+| Action      | Command                                                                                |
++=============+========================================================================================+
+| Resolve     | No Op                                                                                  |
++-------------+----------------------------------------------------------------------------------------+
+| Compile     | ``dotnet lambda package --configuration release --output-package $BUILD/package.zip``  |
++-------------+----------------------------------------------------------------------------------------+
+| Copy Source | No Op                                                                                  |
++-------------+----------------------------------------------------------------------------------------+
+
+
+Python using PIP
+^^^^^^^^^^^^^^^^
+
++-------------+---------------------------------------------------------------------------------------+
+| Action      | Command                                                                               |
++=============+=======================================================================================+
+| Resolve     | ``pip install --isolated --disable-pip-version-check -r requirements.txt -t $BUILD``  |
++-------------+---------------------------------------------------------------------------------------+
+| Compile     | No Op                                                                                 |
++-------------+---------------------------------------------------------------------------------------+
+| Copy Source | Copy all files                                                                        |
++-------------+---------------------------------------------------------------------------------------+
+
+
+Custom Build Commands
+~~~~~~~~~~~~~~~~~~~~~
+
+Users can provide custom build command for each runtime via ``.samrc`` that will be run instead of the default build
+action provided by SAM CLI. SAM CLI will invoke this command for every resource that uses this runtime. This can be any
+arbitrary program that is accessible (via ``PATH`` resolution). The program will be invoked by setting three
```

Comment:
arbitrary screams scary...

Reply:
Yes! I want to skip this feature for MVP because it opens a pandoras box. Running arbitrary commands without a real RPC-like protocol and controls seems scary. At that point customers can just write a full build script themselves. I want to think this through before we go support it.